adapter/internal/adapter/adapter.go

```diff
@@ -226,6 +226,7 @@ func Run(conf *config.Config) {
 	// TODO: (VirajSalaka) Properly configure once the adapter flow is complete.
 	if conf.GlobalAdapter.Enabled {
 		go ga.InitGAClient(conf.GlobalAdapter.ServiceURL)
+		FetchAPIUUIDsFromGlobalAdapter()
```

Comment:
Since this function includes LoadSubscriptionData, it is also triggered twice when GA is enabled?

Reply:
We can remove the LoadSubscriptionData function included here.

Reply:
Fixed with 11f3bf18b5ffcc47bb8dcdd40cda54be1ebe9307

Reply:
Another question. In loadSubscriptionData, the adapter pulls all the available APIs from /apis endpoint. 
IMO we need to restrict that as well.
Simplest way would be passing the apIUUID array and then filter out.

Reply:
Fixed with 11c357a6240dd3f9c34f3bf7b4d7d302cdc59305